src/Helpers/Tokens/IdTokenVerifier.php

```diff
@@ -0,0 +1,245 @@
+        }
+
+        $issuedTime = $tokenIat - $leeway;
+        if ($now < $issuedTime) {
+            throw new InvalidTokenException( sprintf(
+                'Issued At (iat) claim error in the ID token; current time (%d) is before issued at time (%d)',
+                $now,
+                $issuedTime
+            ) );
+        }
+
+        /*
+         * Nonce check
+         */
+
+        if (! empty($options['nonce'])) {
+            $tokenNonce = $verifiedToken->getClaim('nonce', false);
+            if (! $tokenNonce || ! is_string($tokenNonce)) {
+                throw new InvalidTokenException('Nonce (nonce) claim must be a string present in the ID token');
+            }
+
+            if ($tokenNonce !== $options['nonce']) {
+                throw new InvalidTokenException( sprintf(
+                    'Nonce (nonce) claim mismatch in the ID token; expected "%s", found "%s"',
+                    $options['nonce'],
+                    $tokenNonce
+                ) );
+            }
+        }
+
+        /*
+         * Authorized party check
+         */
+
+        if (is_array($tokenAud) && count($tokenAud) > 1) {
+            $tokenAzp = $verifiedToken->getClaim('azp', false);
+            if (! $tokenAzp || ! is_string($tokenAzp)) {
+                throw new InvalidTokenException(
+                    'Authorized Party (azp) claim must be a string present in the ID token when Audience (aud) claim has multiple values'
+                );
+            }
+
+            if ($tokenAzp !== $this->audience) {
+                throw new InvalidTokenException( sprintf(
+                    'Authorized Party (azp) claim mismatch in the ID token; expected "%s", found "%s"',
+                    $this->audience,
+                    $tokenAzp
+                ) );
+            }
+        }
+
+        /*
+         * Authentication time check
+         */
+
+        if (! empty($options['max_age'])) {
+            $tokenAuthTime = $verifiedToken->getClaim('auth_time', false);
+            if (! $tokenAuthTime || ! is_int($tokenAuthTime)) {
+                throw new InvalidTokenException(
+                    'Authentication Time (auth_time) claim must be a number present in the ID token when Max Age (max_age) is specified'
+                );
+            }
+
+            $authValidUntil = $tokenAuthTime + $options['max_age'] + $leeway;
+
+            if ($now > $authValidUntil) {
+                throw new InvalidTokenException( sprintf(
+                    'Authentication Time (auth_time) claim in the ID token indicates that too much time has passed since the last end-user authentication. Current time (%d) is after last auth at %d',
+                    $now,
+                    $authValidUntil
+                ) );
+            }
+        }
+
+        $profile = [];
+        foreach ($verifiedToken->getClaims() as $claim => $value) {
+            $profile[$claim] = $value->getValue();
```

Comment:
do you need to do this or can you just return `getClaims()` ?

Reply:
Because of the `Token` type that's returned being incompatible with the rest of the SDK, yes, unfortunately.